### PR TITLE
Upgrade proto to v1.4.0

### PR DESCRIPTION
**Pull request by florianl**
Release of the [v1.4.0][otlp] version of the OTLP.

[otlp]: https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v1.4.0

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Comment by pellared** [comment]
I have a feeling that a newer version of Go is needed because of go-grpc dependency: https://github.com/open-telemetry/opentelemetry-proto-go/blob/53bebda7f3b0c1d07ea9babf4b049ede28555849/otlp/go.mod#L3

See https://github.com/open-telemetry/opentelemetry-proto-go/pull/194#discussion_r1856297893

**Comment by pellared** [comment]
I think it is a bug that should be fixed before the release.

**Comment by florianl** [comment]
I'm not sure and can't follow how this is related to this PR. From checking the PR again, I didn't identify an element, that would require a Go version bump.
Is there something in the current code base, that would require a Go version bump, was missed and is unrelated to this PR?

**Comment by pellared** [comment]
> I'm not sure and can't follow how this is related to this PR. From checking the PR again, I didn't identify an element, that would require a Go version bump. Is there something in the current code base, that would require a Go version bump, was missed and is unrelated to this PR?

I created a bug: https://github.com/open-telemetry/opentelemetry-proto-go/issues/195

**Comment by dmathieu** [comment]
This should be mergeable, once someone from @open-telemetry/proto-go-approvers approves.

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Versioning of modules in this project will be idiomatic of a Go project using [G
 They will use [semantic import versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning).
 Meaning modules will comply with [semver 2.0](https://semver.org/spec/v2.0.0.html) with the following exception:
 
-- Packages with an `experimental` suffix do not comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
+- Packages with an `experimental` or `development` suffix do not comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
```
[the remark this answers is not in the document]
Comment (florianl) [reply README.md:17]: ok - I will drop this :+1:

**Comment by florianl** [comment]
The following error sounds like a circular dependency issue resulting from multiple `go.mod` files:

```
go: finding module for package go.opentelemetry.io/proto/slim/otlp/profiles/v1development
go: go.opentelemetry.io/proto/gen/go/go.opentelemetry.io/proto/slim/otlp/collector/profiles/v1development imports
	go.opentelemetry.io/proto/slim/otlp/profiles/v1development: module go.opentelemetry.io/proto/slim/otlp@latest found (v1.3.1), but does not contain package go.opentelemetry.io/proto/slim/otlp/profiles/v1development
make: *** [Makefile:169: go-mod-tidy/.] Error 1
```

**Comment by dmathieu** [comment]
From the CI output, it looks like `go-mod-tidy` still doesn't remove the `gen` folder, which is weird as it works fine both on my mac and on a debian VM.

**Comment by pellared** [comment]
You should not generate `slim/otlp/collector/profiles/v1development`. `slim` modules should not have `collector` directory which brings the `grpc` dependency.

**Comment by florianl** [comment]
> You should not generate slim/otlp/collector/profiles/v1development. slim modules should not have collector directory which brings the grpc dependency.
> 
> I wonder how was it even generated...

Can you provide more background on this? the Makefile target `gen-otlp-protobuf-slim` does not differentiate between collector and other parts.
https://github.com/open-telemetry/opentelemetry-proto-go/blob/f83a33f5490808194d466cc3ff18c6e66a48a407/Makefile#L135-L139

While in the target `gen-otlp-protobuf` for every signal also a collector part is generated:
https://github.com/open-telemetry/opentelemetry-proto-go/blob/f83a33f5490808194d466cc3ff18c6e66a48a407/Makefile#L118C9-L126

**Comment by florianl** [comment]
I can reproduce the issue with a clean state on `main` and following the release process:

```
# update Makefile and repalce experimental with development
make sync VERSION=v1.4.0
# update versions.yaml from 1.3.1 to 1.4.0
make verify-versions
git diff
```
It is in all cases the combination of `make clean protobuf go-mod-tidy` ( as executed in CI), that breaks something.

**Comment by pellared** [comment]
> Can you provide more background on this?

https://github.com/open-telemetry/opentelemetry-proto-go/pull/155

Maybe I did some mistake(s). I am not a `make` expert.

**Review by pellared — APPROVED** [review]
@florianl, @dmathieu, good work 👍

**Pull request merged**